scripts/setup-centos9.sh

```diff
@@ -64,7 +60,7 @@ function install_velox_deps_from_dnf {
   dnf_install libevent-devel \
     openssl-devel re2-devel libzstd-devel lz4-devel double-conversion-devel \
     libdwarf-devel elfutils-libelf-devel curl-devel libicu-devel bison flex \
-    libsodium-devel zlib-devel
+    libsodium-devel zlib-devel gmock gtest
```

Comment:
We need the `gmock-devel` here. This pulls in gtest-devel too.

Reply:
Since #10422 we now properly detect system gtest, so having that in the images would be nice anyway.

Reply:
Yeah. Actually, building the Arrow dependency failed for me from a fresh system until I had gtest-devel installed. Arrow has testing turned on...

On a side note: pip isn't used to install `regex` and `cmake-format` anymore? Was that removed?
On a developer machine this would be needed to run `make format-check` / `make-format-fix`. In the CI pipeline it has its own container for checking.
Do we want to add this?

Reply:
See #10668 for the python changes, the CI container for the check job is mostly for clang-format I think because different versions can show very different results.

Reply:
Right, the venv was added for MacOS. So far the python versions on Linux don't cause a problem but we likely will need to expand it. I was just wondering why we install clang-format/regex on MacOS but not the Linux platforms and whether or not we should have it commonly done.

Reply:
Good point actually, not sure? @majetideepak

Reply:
We have a separate script for linux `scripts/setup-check.sh`. This was likely moved out to simplify building the check container.
Developers mainly use MacOS and they need the check set up. Linux is used for deployment and the checks are not required.
We can include setup-check.sh inside the linux scripts and install as well.
We need to ensure Linux and MacOS use the same clang-format version.

Reply:
> Linux is used for deployment and the checks are not required.
I think development on Linux is done. I suppose the developers did the setup themselves. There is an assumption that pip would install the same version on all platforms but it could be fixed.

Let me undo the change though and deal with this later.

Reply:
> Developers mainly use MacOS
eh, maybe (written on linux^^) but even then we want to make it easy to contribute so this should be included in the devsetup for linux as well